src/test/java/com/googlecode/download/maven/plugin/internal/WGetTest.java

```diff
@@ -48,29 +80,195 @@ public void setUp() throws Exception {
         setVariableValueToObject(mojo, "settings", new Settings());
         setVariableValueToObject(mojo, "buildContext", buildContext);
+
+        initializer.accept(mojo);
+        return mojo;
     }
 
     private void cleanUp() {
         if (cacheDirectory.exists()) {
             stream(cacheDirectory.listFiles()).forEach(File::delete);
             cacheDirectory.delete();
         }
-        if (outputFile.exists()) {
-            outputFile.delete();
-        }
     }
 
     @Test
     public void testCacheDirectoryNotCreated()
             throws IllegalAccessException, MojoExecutionException, MojoFailureException {
-        setVariableValueToObject(mojo, "skipCache", true);
+        WGet mojo = createMojo(m -> setVariableValueToObject(m, "skipCache", true));
 
         try {
             mojo.execute();
-            assertThat("Cache directory should remmain empty if skipTest is true", cacheDirectory.list(),
+            assertThat("Cache directory should remain empty if skipTest is true", cacheDirectory.list(),
                     Matchers.emptyArray());
         } finally {
             cleanUp();
         }
     }
+
+    @Test
+    public void testCacheInANonExistingDirectory() throws MojoExecutionException, MojoFailureException, IllegalAccessException {
+        try {
+            // mojo.execute() will produce an error since wagonManager remains null
+            createMojo(m -> {
+                setVariableValueToObject(m, "overwrite", true);
+                File nonExistingDir = mock(File.class);
+                when(nonExistingDir.getAbsolutePath()).thenReturn("/nonExistingDirectory");
+                when(nonExistingDir.exists()).thenReturn(false);
+                setVariableValueToObject(m, "cacheDirectory", nonExistingDir);
+            }).execute();
+            assertThat("Cache directory should remain empty if quit abruptly", cacheDirectory.list(),
+                    Matchers.emptyArray());
+        } finally {
+            cleanUp();
+        }
+    }
+
+    @Test
+    public void testCacheInNotADirectory() throws MojoExecutionException, IllegalAccessException {
+        try {
+            createMojo(m -> {
+                setVariableValueToObject(m, "overwrite", true);
+                File notADirectory = mock(File.class);
+                when(notADirectory.getAbsolutePath()).thenReturn("/nonExistingDirectory");
+                when(notADirectory.exists()).thenReturn(true);
+                when(notADirectory.isDirectory()).thenReturn(false);
+                setVariableValueToObject(m, "cacheDirectory", notADirectory);
+            }).execute();
+            fail();
+        } catch (MojoFailureException e) {
+            assertThat(e.getMessage(), containsString("cacheDirectory is not a directory"));
+        } finally {
+            cleanUp();
+        }
+    }
+
+    @Test
+    public void testCacheNotWrittenToIfFailed() throws MojoExecutionException, MojoFailureException, IllegalAccessException {
```

Comment:
@ajarmoniuk it would be nice to have a javadoc explaining the case. Its not clear from the test method name